pkg/certificate/controller.go

```diff
@@ -33,9 +38,38 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return errors.Wrap(err, "failed instanciating certificate controller")
 	}
 
-	isWebhookConfigOrAnnotatedResource := func(meta metav1.Object) bool {
+	isACascadeDelete := func(deleteEvent event.DeleteEvent) bool {
+		namespace := deleteEvent.Meta.GetNamespace()
+		for _, ownerRefernce := range deleteEvent.Meta.GetOwnerReferences() {
+			gv, err := schema.ParseGroupVersion(ownerRefernce.APIVersion)
+			if err != nil {
+				logger.Error(err, "failed parsing gv from owner")
+				continue
+			}
+			gvk := gv.WithKind(ownerRefernce.Kind)
+			object := unstructured.Unstructured{}
+			object.SetGroupVersionKind(gvk)
+			err = m.client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: ownerRefernce.Name}, &object)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					logger.Info("Cascade delete detected, at least one owner is not present")
```

Comment:
Maybe it would be easier to understand if we called it "garbage collection detected"? It is more known than "cascade"